CMakeLists.txt

```diff
@@ -323,15 +324,36 @@ if(USE_CCACHE)
   endif()
 endif()
 
-# Build type configuration - default to Release if none is set
+# Build type configuration - default to Release if none is set unless
+# code coverage is in use, in which case override to Debug
 
-if(NOT CMAKE_BUILD_TYPE)
+if (OPENVDB_CODE_COVERAGE)
+  message(WARNING "Code coverage requires debug mode, setting CMAKE_BUILD_TYPE "
+    "to Debug.")
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING
+    "Code coverage requires debug mode." FORCE
+  )
+elseif(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release CACHE STRING
     "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." FORCE
   )
 endif()
 message(STATUS "CMake Build Type: ${CMAKE_BUILD_TYPE}")
 
+# Code coverage configuration
+
+if(OPENVDB_CODE_COVERAGE)
+  # set gcov compile and link flags
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
+
+  # use .gcno extension instead of .cc.gcno
+  set(CMAKE_CXX_OUTPUT_EXTENSION_REPLACE 1)
```

Comment:
Why is this necessary?

Reply:
It's common practice using CMake and code coverage to handle out-of-source builds:

https://texus.me/2015/09/06/cmake-and-gcov/

Reply:
Hmm I dunno how common this is, can't find any docs for that variable! But seems like the only solution